### ISSUE TITLE
False positive (and outdated) E0202

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8109]
## Environment

* **IntelliJ Rust plugin version:** 0.4.160.4245-213-nightly
* **Rust toolchain version:** 1.58.0-nightly (547a6ffee 2021-10-21) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 Beta (CL-213.5744.5)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Plugin detect E0202 here, but this code is correct 
<img width="566" alt="изображение" src="https://user-images.githubusercontent.com/49211026/141757057-796aa0eb-020f-4570-90d4-902fd5033acf.png">

And E0202 has been removed from the list of rust errors https://doc.rust-lang.org/error-index.html

## Steps to reproduce
```rust
#![feature(inherent_associated_types)]
struct Foo;
impl Foo {
    type Long = i64;
}

```
